exercises/react/source/react/ReactorTest.ceylon

```diff
@@ -0,0 +1,137 @@
+import ceylon.test { ... }
+
+test
+void inputCellsHaveValue() {
+  value r = Reactor();
+  value input = r.newInputCell(10);
+  assertEquals(input.currentValue, 10);
+}
+
+test
+void inputCellsCanHaveValuesSet() {
+  value r = Reactor();
+  value input = r.newInputCell(4);
+  input.currentValue = 20;
+  assertEquals(input.currentValue, 20);
+}
+
+test
+void computeCellsCalculateInitialValue() {
+  value r = Reactor();
+  value input = r.newInputCell(1);
+  value output = r.newComputeCell1(input, (x) => x + 1);
+  assertEquals(output.currentValue, 2);
+}
+
+test
+void computeCellsTakeInputInRightOrder() {
+  value r = Reactor();
+  value one = r.newInputCell(1);
+  value two = r.newInputCell(2);
+  value output = r.newComputeCell2(one, two, (x, y) => x + y * 10);
+  assertEquals(output.currentValue, 21);
+}
+
+test
+void computeCellsUpdateValueWhenDependenciesChange() {
+  value r = Reactor();
+  value input = r.newInputCell(1);
+  value output = r.newComputeCell1(input, (x) => x + 1);
+  input.currentValue = 3;
+  assertEquals(output.currentValue, 4);
+}
+
+test
+void computeCellsCanDependOnOtherComputeCells() {
+  value r = Reactor();
+  value input = r.newInputCell(1);
+  value timesTwo = r.newComputeCell1(input, (x) => x * 2);
+  value timesThirty = r.newComputeCell1(input, (x) => x * 30);
+  value output = r.newComputeCell2(timesTwo, timesThirty, (x, y) => x + y);
+
+  assertEquals(output.currentValue, 32);
+  input.currentValue = 3;
+  assertEquals(output.currentValue, 96);
+}
+
+test
+void computeCellsFireCallbacks() {
+  value r = Reactor();
+  value input = r.newInputCell(1);
+  value output = r.newComputeCell1(input, (x) => x + 1);
+
+  variable Element[] vals = [];
+  output.addCallback((x) => vals = vals.withTrailing(x));
+
+  input.currentValue = 3;
+  assertEquals(vals, [4]);
+}
+
+test
+void callbacksOnlyFireOnChange() {
+  value r = Reactor();
+  value input = r.newInputCell(1);
+  value output = r.newComputeCell1(input, (x) => if (x < 3) then 111 else 222);
+
+  variable Element[] vals = [];
+  output.addCallback((x) => vals = vals.withTrailing(x));
+
+  input.currentValue = 2;
+  assertEquals(vals, []);
+
+  input.currentValue = 4;
+  assertEquals(vals, [222]);
+}
+
+test
+void removingCallbackMultipleTimesDoesntInterfereWithOtherCallbacks() {
```

Comment:
there was supposed to be another test saying that callbacks can be removed - where did it go??????